### PR TITLE
Collection and generics.

### PR DESCRIPTION
**Pull request by dceaicovschi**

### CODE REVIEW

File: src/main/java/com/endava/soa_5/collection/User.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public void setAddresses(String key, Address address) {
+        this.userAddresses.put(key, address);
+        System.out.println();
+    }
+
+    public boolean isUserFrom(String city) {
+        for (Address address : userAddresses.values()) {
+            if (address.getCity().equals(city)) {
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/collection/User.java]: Can be there lower/upper case mismatch ?

File: src/main/java/com/endava/soa_5/generic/GenericArray.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+public class GenericArray {
+    public static <T> void printList(T[] list) {
+        for (T value : list) {
+            System.out.print(value + " ");
+        }
+        System.out.println();
+    }
+
+    public static <Z> Z[] insertString(Z[] list) {
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/generic/GenericArray.java]: What is the scope of this method ?

File: src/main/java/com/endava/soa_5/collection/Main.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        order12.addItem(item11);
+
+        List<Order> orders = new ArrayList<>();
+        orders.addAll(Arrays.asList(order1, order2, order3, order4, order5, order6));
+        orders.addAll(Arrays.asList(order7, order8, order9, order10, order11, order12));
+        System.out.println(filterItemsByType(orders, Type.TOY));
+
+        List<User> users = new ArrayList<>();
+        users.addAll(Arrays.asList(user1, user2, user3));
+//        filteringUsersByCity(users,"Chisinau");
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/collection/Main.java]: Try to avoid commented code lines

Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/collection/Main.java]: Done!

File: src/main/java/com/endava/soa_5/collection/User.java (hunk 1 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/collection/User.java]: Done, values are compared in value.UpperCase();

File: src/main/java/com/endava/soa_5/generic/GenericArray.java (hunk 2 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/generic/GenericArray.java]: Was deleted. unused method!

File: src/main/java/com/endava/soa_5/collection/User.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public void setAddresses(String key, Address address) {
+        this.userAddresses.put(key, address);
+        System.out.println();
+    }
+
+    public boolean isUserFrom(String city) {
+        for (Address address : userAddresses.values()) {
+            if (address.getCity().toUpperCase().equals(city.toUpperCase())) {
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/collection/User.java:33]: It's ok, but can use equalsIgnoreCase()

**Review by MarcelMorozniuc — APPROVED** [review]

**Review by margaritaq — APPROVED** [review]

File: src/main/java/com/endava/soa_5/collection/User.java (hunk 4 above)
Comment (dceaicovschi) [reply src/main/java/com/endava/soa_5/collection/User.java:33]: Ok!, Thanks for the hint, will do next time!

**Pull request merged**